### PR TITLE
Fix for the SVG SubPath Issue.

### PR DESCRIPTION
**Pull request by Emasoft**
Fix for the SVG SubPath Issue. Now subpaths are rendered correctly.

### CODE REVIEW

**Comment by Emasoft** [comment]
Subpath bug BEFORE the Fix:
subpath bug before fix

Subpath bug AFTER the Fix
subpath bug fixed after fix

**Comment by BlueRaja** [comment]
Diff without whitespace changes: https://github.com/praeclarum/NGraphics/pull/62/files?w=1

File: NGraphics/SvgReader.cs [hunk 1]
```diff
@@ -9,978 +9,991 @@
...
+
+            if (widthA != null && widthA.Value.Contains ("%")) {
+                size.Width *= viewBox.Width;
+            }
+            if (heightA != null && heightA.Value.Contains ("%")) {
+                size.Height *= viewBox.Height;
+            }
+
+            System.Diagnostics.Debug.WriteLine ("SVG viewbox: " + viewBox.ToString ());
+            System.Diagnostics.Debug.WriteLine ("SVG size: " + size.ToString ());
```
Comment (BlueRaja) [inline NGraphics/SvgReader.cs]: Left in by mistake?

Comment (Emasoft) [inline NGraphics/SvgReader.cs]: Yes, sorry

Comment (Emasoft) [inline NGraphics/SvgReader.cs]: Can you remove those lines and merge anyway? Or I need to do another PR?

Comment (BlueRaja) [inline NGraphics/SvgReader.cs]: I'm not the maintainer of this repo, I'm just watching it.  But you can fix it yourself by submitting another commit to the same branch. The PR will update automatically.

Comment (Emasoft) [inline NGraphics/SvgReader.cs]: Thanks, I'll do.

Comment (Emasoft) [inline NGraphics/SvgReader.cs]: Ok, debug code removed!

**Comment by Emasoft** [comment]
@praeclarum Frank can you merge this, please? I need to share a project on GitHub with those fixes and I don't want to create a fork with the fix and a separate Nuget.

**Comment by praeclarum** [comment]
I don't like accepting changes with all these whitespace breaks - ruins history and blame.

**Comment by BlueRaja** [comment]
To be fair, the [official Microsoft C# coding guidelines](https://msdn.microsoft.com/en-us/library/ff926074.aspx), followed by the vast majority of C# developers, recommends using 4-space indents.

**Comment by praeclarum** [comment]
[Contribution Etiquette](http://tirania.org/blog/archive/2010/Dec-31.html)

Anyway, these are good changes. If @Emasoft doesn't want to fix this PR then I can use the diffs to hand merge this puppy. I just don't like that @Emasoft would lose the credit.

**Comment by Emasoft** [comment]
I would love to fix it, but I don't know how. I just loaded the project in Xamarin Studio, and it uses the Visual Studio code formatting settings by default:

schermata 2016-11-30 alle 20 26 25
schermata 2016-11-30 alle 20 25 48
 
Is there a unix/osx **shell script** I can use to revert the .cs file to your formatting preferences? 

What formatting standard are you using? Is there a specific **formatting policy file** to use for this project? A mandatory policy file to use would be great for avoiding messing up the formatting.

If you use a specific formatting policy, you can export them from _Xamarin Studio -> Policies.. -> Export_ , like this:

schermata 2016-11-30 alle 20 35 35

If you export it, I can load it in Xamarin Studio and apply it to the source code. Or you can do this yourself as an additional commit of this pull request. If it is impossible to do, then hand merge it. I'm not happy to lose the credit but at least it will be merged. Thank you!

**Comment by praeclarum** [comment]
The problem isn't that your formatting is wrong, it's that you reformatted the whole file. Simply don't do that.

You must have a setting somewhere that reformats the whole document on change. Just turn that off or don't hit that command.

I am not using a formatting policy. Please just look at the file and match what is already being done.

File: NGraphics/SvgReader.cs [hunk 2]
```diff
@@ -425,10 +425,14 @@ protected GradientBrush GetGradientBrush(string fill, GradientBrush child)
                         ((RadialGradientBrush)brush).Center = ((RadialGradientBrush)child).Center;
                         ((RadialGradientBrush)brush).Focus = ((RadialGradientBrush)child).Focus;
                         ((RadialGradientBrush)brush).Radius = ((RadialGradientBrush)child).Radius;
+                        ((RadialGradientBrush)brush).Absolute = ((RadialGradientBrush)child).Absolute;
+                        ((RadialGradientBrush)brush).gradientTransform = ((RadialGradientBrush)child).gradientTransform;
```
Comment (praeclarum) [inline NGraphics/SvgReader.cs]: `gradientTransform` should be capitalized

**Comment by Emasoft** [comment]
I have the option "Xamarin Studio -> Preferences -> Behaviour -> Format document on save.. " on, because I like my code to be always formatted.  I will switch this off in the future. But for now is there a way to fix this applying a new formatting scheme like the one you use?

File: NGraphics/SvgReader.cs (hunk 2 above)
Comment (Emasoft) [inline NGraphics/SvgReader.cs]: You are right. I will fix this asap and add a new commit to this PR.

**Comment by praeclarum** [comment]
Yeah, sorry this project *should* have a policy in the solution so you can have your auto format be safe. I'll look into this...

I'm afraid the only way to fix this is to rebase from master and apply the changes by hand again.

Sorry for trouble. I do appreciate the work. If you're unwilling to do it, I can merge myself.

**Comment by BlueRaja** [comment]
@praeclarum I don't disagree with you, I was just playing Devil's Advocate
